### PR TITLE
[interoperator] Kubebuilder 2.0

### PR DESCRIPTION
**Pull request by vivekzhere**
- Initialize kubebuilder 2.0 interoperator
- Migrate Schedulers
- Migrate Provisioner
- Migrate MCD
- Move /pkg/internal/* to /internal

Feature: HCPCFS-2473

TODO
Update docs and DockerFile

### CODE REVIEW

File: interoperator/Dockerfile [hunk 1]
```diff
@@ -1,24 +1,34 @@
...
+COPY api/ api/
+COPY controllers/ controllers/
+COPY internal/ internal/
+COPY pkg/ pkg/
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -tags provisioners -a -o manager github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/cmd/manager
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -tags schedulers -a -o scheduler github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/cmd/manager
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -tags multiclusterdeploy -a -o multiclusterdeploy github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/cmd/manager
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -tags provisioners -a -o manager main.go
```
Comment (subhankarc) [inline interoperator/Dockerfile]: Why is this `GO111MODULE=on` required here? We do not have it in the GOPATH, right ?

File: interoperator/Dockerfile [hunk 2]
```diff
@@ -1,24 +1,34 @@
...
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -tags schedulers -a -o scheduler main.go
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -tags multiclusterdeploy -a -o multiclusterdeploy main.go
 
-# Copy the controller-manager into a thin image
-FROM ubuntu:latest
+# Use distroless as minimal base image to package the manager binary
+# Refer to https://github.com/GoogleContainerTools/distroless for more details
+FROM gcr.io/distroless/static:nonroot
```
Comment (subhankarc) [inline interoperator/Dockerfile]: Any reason why we used distroless image ?

File: interoperator/Makefile [hunk 3]
```diff
@@ -1,70 +1,86 @@
 
 # Image URL to use all building/pushing image targets
 IMG ?= controller:latest
+# Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
```
Comment (subhankarc) [inline interoperator/Makefile]: Why is this done?

File: helm-charts/interoperator/templates/sfcluster.yaml [hunk 4]
```diff
@@ -2,17 +2,18 @@ apiVersion: apiextensions.k8s.io/v1beta1
...
   name: sfclusters.resource.servicefabrik.io
 spec:
   group: resource.servicefabrik.io
   names:
     kind: SFCluster
+    listKind: SFClusterList
     plural: sfclusters
-  scope: Namespaced
+    singular: sfcluster
+  scope: ""
```
Comment (subhankarc) [inline helm-charts/interoperator/templates/sfcluster.yaml]: `sfcluser` is no more namespaced scoped is it ?

File: .travis.yml [hunk 5]
```diff
@@ -5,21 +5,21 @@ matrix:
       os:
         - linux
       go:
-        - "1.11"
```
[the remark this answers is not in the document]
Comment (vivekzhere) [reply .travis.yml]: Works with go 1.11 also.

File: helm-charts/interoperator/templates/sfcluster.yaml (hunk 4 above)
Comment (vivekzhere) [reply helm-charts/interoperator/templates/sfcluster.yaml]: "" means Namespaced as it is the default value. So all our resources are still namespace scoped. This change is due to kubebuilder. The file is generated

**Pull request merged**